### PR TITLE
Fix behavior #url_for with string as argument

### PR DESCRIPTION
**Pull request by bogdanvlviv**
Fix inconsistency between String and Symbol #url_for handling

Related to 37d4415a7b433fcb987b1c6a5b51bf2d8efc5d5e

/cc @tenderlove

### CODE REVIEW

**Comment by maclover7** [comment]
@bogdanvlviv please take a look at the failing build on Travis

File: actionpack/lib/action_dispatch/routing/url_for.rb [hunk 1]
```diff
@@ -165,28 +165,36 @@ def url_options
       # last +url_for+ calls.
       def url_for(options = nil)
         case options
-        when nil
-          _routes.url_for(url_options.symbolize_keys)
+        when String
+          if options =~ /[\/\#\?]/
```
Comment (bogdanvlviv) [inline actionpack/lib/action_dispatch/routing/url_for.rb]: I'm not sure about this. I need feedback, please.

**Comment by tenderlove** [comment]
@bogdanvlviv my point behind adding that test is to ensure that the behavior doesn't regress.  It's not right to make the behavior consistent without showing *why* it behaves the way it does today.

I don't think it's a good idea to merge this PR unless the following questions can be answered:

1. Is the current behavior merely a bug, or is it intentional?
2. Can you find where it was introduced?
3. Is there a reason we *need* to make this change?

If it's merely a bug, then answering 3 is no problem.

**Comment by bogdanvlviv** [comment]
@tenderlove 
1.

In my opinion should be some consistency (but i can ), when we see

```ruby
url_for("projects") # => "projects"
```

```ruby
url_for(:projects) # =>"http://example.com/projects"
```
we are expecting same result.

It think when `#url_for` receives `string` argument, and `string.to_sym` isn't "dyna symbol"
it should work same as with `symbol`.

> I need help with checking on "dyna symbol" here.

2. No.

3. I made this PR for discussing it.

**Comment by bogdanvlviv** [comment]
#28073

**Pull request closed without merging**